sal-service/src/main/java/org/ow2/proactive/sal/service/util/ConnectionHelper.java

```diff
@@ -71,17 +76,51 @@ public static JSONObject sendGetRequestAndReturnObjectResponse(URI requestUri) {
     }
 
     @SneakyThrows
-    public static JSONArray sendGetArrayRequestAndReturnArrayResponse(URI requestUri) {
-        HttpURLConnection connection = (HttpURLConnection) requestUri.toURL().openConnection();
-        connection.setRequestMethod(HttpMethod.GET.toString());
-        LOGGER.debug("requestUri = {}", requestUri);
-
-        BufferedReader br = sendGetRequestAndReturnBufferedResponse(connection);
-
-        JSONArray result = (br != null) ? new JSONArray(new JSONTokener(br)) : null;
-
-        connection.disconnect();
+    public static JSONArray sendGetArrayRequestAndReturnArrayResponse(URI requestUri) throws IOException {
+        HttpURLConnection connection = null;
+        BufferedReader br = null;
+        JSONArray result = new JSONArray();
+
+        try {
+            connection = (HttpURLConnection) requestUri.toURL().openConnection();
+            connection.setRequestMethod(HttpMethod.GET.toString());
```

Comment:
A minor comment. At this point, we can check whether the connection is null, before going further with a null connection.